reason-react-native/src/components/DrawerLayoutAndroid.re

```diff
@@ -29,7 +29,7 @@ external make:
                        =?,
     ~onDrawerOpen: unit => unit=?,
     ~onDrawerSlide: drawerSlideEvent => unit=?,
-    ~onDrawerStateChanged: [ | `Idle | `Dragging | `Settling] => unit=?,
+    ~onDrawerStateChanged: AndroidInteractionState.t => unit=?,
```

Comment:
Yes, as it is a callback param, bs.string won't work here.

But why is it called AndroidInteractionState? Flow has

```js
type DrawerStates = 'Idle' | 'Dragging' | 'Settling';
```

The plural is weird, but I would just call it DrawerLayoutAndroid.DrawerState.t.

Reply:
For `ViewPagerAndroid` they are called `PageScrollState`. I'm aware it will be removed from RN core, but still it's useful to have.

Instead of two identical type definitions, I figured a common one would work better.

Reply:
Hmm, ok - what do you think @MoOx?

Reply:
Works for me the way it is, anyway this types are kind of internal types (users won't directly use it except for edge cases?) so no big deal if we want to revisit later.